src/osd/OSD.h

```diff
@@ -1175,6 +1175,29 @@ class OSDService {
   }
 #endif
 
+  void check_completion_item();
+#if 0
+  class CompletionThread : public Thread {
+  public:
+    OSD * osd;
+    CompletionThread(OSD *osd) : osd(osd) {}
+    void *entry() {
+      osd->check_completion_item();
+      return 0;
+    }
+  }comp_thread;
+#endif
+  class CompletionThread : public Thread {
+  public:
+    OSDService * osd_srv;
+    CompletionThread(OSDService *osd_srv) : osd_srv(osd_srv) {}
+    void *entry() {
+      osd_srv->check_completion_item();
+      return 0;
+    }
+  }comp_thread;
+
+  bool comp_thread_stop;
```

Comment:
@athanatos Actually, this is checking whether thread is stop or not. I don't think multiple threads access this value because stop is set rarely. other threads in OSD use same approach (For example, agent thread and heartbeat thread..)